gpytorch/metrics/metrics.py

```diff
@@ -0,0 +1,73 @@
+from math import pi
+
+import torch
+
+from ..distributions import MultivariateNormal
+
+pi = torch.tensor(pi)
+
+
+def negative_log_predictive_density(
+    pred_dist: MultivariateNormal,
+    test_y: torch.Tensor,
+):
+
+    with torch.no_grad():
+        return -pred_dist.log_prob(test_y).item()
```

Comment:
```suggestion
        return -pred_dist.log_prob(test_y)
```

Reply:
Also, would suggest returning a normalization by `test_y.shape[-1]` here as well

Reply:
Does that mean normalizing by the number of samples? Also, batch support would be desirable here?

Reply:
yes, normalizing by number of samples. this should return a batch if test_y is batched already.

Reply:
I am trying to create a separate test case for MultiTaskGP where we can have `test_y` batched. It seems that NLPD returns a scaler in this case. I'm not sure if it is obvious because of my ignorance of Multi-Task GPs. The model I used is the following from [this notebook](https://docs.gpytorch.ai/en/stable/examples/03_Multitask_Exact_GPs/Multitask_GP_Regression.html):

```py
class ExactMultiTaskGPModel(ExactGP):
    def __init__(self, train_x, train_y, likelihood):
        super().__init__(train_x, train_y, likelihood)
        self.mean_module = gpytorch.means.MultitaskMean(
            gpytorch.means.ConstantMean(), num_tasks=2
        )
        self.covar_module = gpytorch.kernels.MultitaskKernel(
            gpytorch.kernels.RBFKernel(), num_tasks=2, rank=1
        )

    def forward(self, x):
        mean_x = self.mean_module(x)
        covar_x = self.covar_module(x)
        return gpytorch.distributions.MultitaskMultivariateNormal(mean_x, covar_x)
```

Reply:
Resolved by updated unit test and the following stack of commits.